internal/tofu/context_import.go

```diff
@@ -71,76 +72,80 @@ func (i *ImportTarget) IsFromImportCommandLine() bool {
+		return &i.CommandLineImportTarget.Addr
 	} else {
-		// TODO change this later, when Config.To is not a static address
-		address = i.Config.To
+		return i.Config.ResolvedTo
 	}
-	return
-}
-
-// ResolvedConfigImportsKey is a key for a map of ImportTargets originating from the configuration
-// It is used as a one-to-one representation of an EvaluatedConfigImportTarget.
-// Used in ImportResolver to maintain a map of all resolved imports when walking the graph
-type ResolvedConfigImportsKey struct {
-	// An address string is one-to-one with addrs.AbsResourceInstance
-	AddrStr string
-	ID      string
 }
 
 // ImportResolver is a struct that maintains a map of all imports as they are being resolved.
 // This is specifically for imports originating from configuration.
 // Import targets' addresses are not fully known from the get-go, and could only be resolved later when walking
 // the graph. This struct helps keep track of the resolved imports, mostly for validation that all imports
-// have been addressed and point to an actual configuration
+// have been addressed and point to an actual configuration.
+// The key of the map is a string representation of the address, and the value is an EvaluatedConfigImportTarget.
 type ImportResolver struct {
 	mu      sync.RWMutex
-	imports map[ResolvedConfigImportsKey]EvaluatedConfigImportTarget
+	imports map[string]EvaluatedConfigImportTarget
 }
 
 func NewImportResolver() *ImportResolver {
-	return &ImportResolver{imports: make(map[ResolvedConfigImportsKey]EvaluatedConfigImportTarget)}
+	return &ImportResolver{imports: make(map[string]EvaluatedConfigImportTarget)}
 }
 
 // ResolveImport resolves the ID and address (soon, when it will be necessary) of an ImportTarget originating
 // from an import block, when we have the context necessary to resolve them. The resolved import target would be an
 // EvaluatedConfigImportTarget.
 // This function mutates the EvalContext's ImportResolver, adding the resolved import target
 // The function errors if we failed to evaluate the ID or the address (soon)
-func (ri *ImportResolver) ResolveImport(importTarget *ImportTarget, ctx EvalContext) error {
-	importId, evalDiags := evaluateImportIdExpression(importTarget.Config.ID, ctx)
+func (ri *ImportResolver) ResolveImport(importTarget *ImportTarget, ctx EvalContext) tfdiags.Diagnostics {
+	// The import block expressions are declared within the root module.
+	// We need to explicitly use the context with the path of the root module, so that all references will be
+	// relative to the root module
+	rootCtx := ctx.WithPath(addrs.RootModuleInstance)
+
+	importId, evalDiags := evaluateImportIdExpression(importTarget.Config.ID, rootCtx)
 	if evalDiags.HasErrors() {
-		return evalDiags.Err()
+		return evalDiags
 	}
 
-	// TODO - Change once an import target's address is more dynamic
-	importAddress := importTarget.Config.To
+	importAddress, addressDiags := rootCtx.EvaluateImportAddress(importTarget.Config.To)
```

Comment:
This should be merging the evalDiags with addressDiags.  The code as written will skip warnings.  This comment also applies to the for loop

Reply:
Fixed [Ensure ResolveImport code will support passing warnings](https://github.com/opentofu/opentofu/pull/1270/commits/78417b862bdc20cd4fbdbd359b35d40b751a55c0)